### ISSUE TITLE
teslaCtrlSendCommand; Failed to execute command: car could not execute command: SetChargingAmps failed

### ISSUE THREAD

**Issue by chrispazz** [tesla-local-control/tesla_ble_mqtt_docker#51]
Hello!

Since 2024.26.7 I have no way to get under 5 Amps (from car, from tesla app and from ble)....

Anyone with same problem?

Chris

**Comment by iainbullock** [comment]
Still works ok for me. What do the logs say? Here's mine when setting to 1 amp:

2024-08-19T12:45:05.844906497Z Command charging-set-amps 5 was successfully delivered to vin:LRXXXXXXXXX
2024-08-19T12:45:14.282956788Z Received MQTT message; topic:tesla_ble/LRXXXXXXXXX/charging-set-amps msg:1 vin:LRXXXXXXXXXcmd:charging-set-amps
2024-08-19T12:45:14.287278892Z Attempt 1/5 sending Set charging Amps to 1 (#1/2) to vin:LRXXXXXXXXXcommand:charging-set-amps 1
2024-08-19T12:45:17.597807879Z Command charging-set-amps 1 was successfully delivered to vin:LRXXXXXXXXX
2024-08-19T12:45:19.606899653Z Attempt 1/5 sending Set charging Amps to 1 (#2/2) to vin:LRXXXXXXXXXcommand:charging-set-amps 1
2024-08-19T12:45:22.876516988Z Command charging-set-amps 1 was successfully delivered to vin:LRXXXXXXXXX

**Comment by chrispazz** [comment]
Only this for every retry it does:

Received MQTT message; topic:tesla_ble/XXXXXXXXXXXXX/charging-set-amps msg:1 vin:XXXXXXXXXXXXX cmd:charging-set-amps
Attempt 1/5 sending Set charging Amps to 5A then to 1 to vin:XXXXXXXXXXXXXXXXXX command:charging-set-amps 1
teslaCtrlSendCommand; Failed to execute command: car could not execute command: SetChargingAmps failed
Skipping command charging-set-amps 1 to vin:XXXXXXXXXXXXXXXXX

**Comment by iainbullock** [comment]
Maybe Tesla have changed something which would be very annoying. I'm on 2024.20.9

**Comment by chrispazz** [comment]
I tried also with debug enabled but the message is the same.
However I noticed sometimes various

teslaCtrlSendCommand; Error: ble: failed to enumerate device services: ATT request failed: input channel closed: io: read/write on closed pipe

**Comment by iainbullock** [comment]
How far away is your device from the car? The error message you report is more frequent if the distance is further

**Comment by chrispazz** [comment]
Yes, I was suspecting the same....it is not so near the car... :)

**Comment by iainbullock** [comment]
Any way you can try it closer? At least to rule out or otherwise whether Tesla have changed anything

**Comment by chrispazz** [comment]
Not able, the raspberry I am using is mounted inside an electrical box....
However it was working good before last tesla update.... Let's hope is not a Tesla limit.

**Comment by iainbullock** [comment]
OK. I'll see if I can upgrade my car's software to see what happens

**Comment by baylanger** [comment]
```teslaCtrlSendCommand; Error: ble: failed to enumerate device services: ATT request failed: input channel closed: io: read/write on closed pipe```

That error tells us Linux probably can't communicate with the BLE adapter. Please reboot your device and see if it starts working again.

**Comment by raphmur** [comment]
Working fine for me with latest car update. @chrispazz do you still have the issue?

**Comment by chrispazz** [comment]
Resolved today. I was a Tesla side issue.
Resolved using this solution:

https://github.com/teslamotors/vehicle-command/issues/287#issuecomment-2306575740

**Comment by iainbullock** [comment]
Thanks for the update, glad it got resolved
